bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tgraph.py

```diff
@@ -34,9 +37,51 @@ def set_size(self, buf):
 # The pythonization consists in setting the size of the array that the getter
 # method returns, so that it is known in Python and the array is fully usable
 # (its length can be obtained, it is iterable).
-comp = cppyy.py.compose_method('^TGraph(2D)?$|^TGraph.*Errors$', # class to match
-                               'GetE?[XYZ]$',                    # method to match
-                               set_size)                         # post-process function
+comp = cppyy.py.compose_method(
+    "^TGraph(2D)?$|^TGraph.*Errors$", "GetE?[XYZ]$", set_size  # class to match  # method to match
+)  # post-process function
 
 # Add the composite to the list of pythonizors
 cppyy.py.add_pythonization(comp)
+
+
+def _init(self, *args):
+    import numpy as np
+
+    if isinstance(args[0], (np.ndarray, np.generic)):
+        self.__class__.__init__(args[0].size, *args)
```

Comment:
Here you should invoke the original implementation of `__init__` that you previously saved, otherwise you are just calling again `_init`.

Also, what happens if args[0] is not a numpy array? You still need to call the original `__init__`.